projects/steering/src/steering_control_stalk.c

```diff
@@ -0,0 +1,55 @@
+#include "steering_control_stalk.h"
+#include <stdio.h>
+#include "adc.h"
+#include "adc_periodic_reader.h"
+#include "can.h"
+#include "can_ack.h"
+#include "can_fifo.h"
+#include "can_fsm.h"
+#include "can_hw.h"
+#include "can_msg.h"
+#include "can_rx.h"
+#include "can_transmit.h"
+#include "event_queue.h"
+#include "exported_enums.h"
+#include "fsm.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "status.h"
+#include "steering_can.h"
+#include "steering_digital_input.h"
+#include "steering_events.h"
+
+// Will be edited later
+#define STEERING_CONTROL_STALK_LEFT_VOLTAGE 1000
+#define STEERING_CONTROL_STALK_RIGHT_VOLTAGE 2000
+#define STEERING_CC_INCREASE_SPEED_VOLTAGE 2000
+#define STEERING_CC_DECREASE_SPEED_VOLTAGE 2500
+#define STEERING_CC_BRAKE_PRESSED_VOLTAGE 100
+

```

Comment:
Your module needs to keep state of where the stalk last was, and only raise events when that changes... right now if the driver turns the signal stalk down to indicate a "Left Signal", this module will bombard `main` with `STEERING_CONTROL_STALK_EVENT_LEFT` events, which isn't good...

Reply:
Make sure you write a test that makes sure this doesn't happen...